pkg/virt-launcher/virtwrap/converter/converter_test.go

```diff
@@ -3277,6 +3277,86 @@ var _ = Describe("Converter", func() {
 				Expect(domain.Spec.Memory.Unit).To(Equal("b"))
 				Expect(domain.Spec.Memory.Value).To(Equal(uint64(guestMemory.Value())))
 			})
+
+			DescribeTable("should correctly convert memory configuration from VMI spec to domain",
+				func(domainSpec v1.DomainSpec, expectedMemoryMiB int64) {
+
+					vmi := &v1.VirtualMachineInstance{
+						ObjectMeta: k8smeta.ObjectMeta{
+							Name:      "testvmi",
+							Namespace: "mynamespace",
+						},
+						Spec: v1.VirtualMachineInstanceSpec{
+							Domain: domainSpec,
+						},
+					}
+
+					v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+
+					testContext := &ConverterContext{
+						VirtualMachine: vmi,
+						AllowEmulation: true,
+						Architecture:   archconverter.NewConverter(runtime.GOARCH),
+					}
+
+					apiDomainSpec := vmiToDomainXMLToDomainSpec(vmi, testContext)
+
+					expectedBytes := expectedMemoryMiB * 1024 * 1024
+					Expect(apiDomainSpec.Memory.Value).To(Equal(uint64(expectedBytes)),
+						"Memory value should be %d bytes (%d MiB)", expectedBytes, expectedMemoryMiB)
+					Expect(apiDomainSpec.Memory.Unit).To(Equal("b"))
+				},
+				Entry("provided by domain spec directly (guest memory takes precedence over limits)",
+					func() v1.DomainSpec {
+						guestMem := resource.MustParse("512Mi")
+						return v1.DomainSpec{
+							Resources: v1.ResourceRequirements{
+								Requests: k8sv1.ResourceList{},
+								Limits:   k8sv1.ResourceList{},
+							},
+							Memory: &v1.Memory{
+								Guest: &guestMem,
+							},
+						}
+					}(),
```

Comment:
did you consider using the more succinct `libvmi`?

If you let the Table accept libvmi.Options, I think this can be changed to `WithGuestMemory("512Mi")`